replica/table.cc

```diff
@@ -3420,9 +3420,9 @@ compaction::table_state& compaction_group::as_table_state() const noexcept {
     return *_table_state;
 }
 
-compaction::table_state& table::as_table_state() const noexcept {
+compaction::table_state& table::as_table_state(size_t id) const noexcept {
     // FIXME: kill it once we're done with all remaining users.
-    return get_compaction_group(0)->as_table_state();
+    return get_compaction_group(id)->as_table_state();
```

Comment:
Is this the only remaining user of this function?
As the FIXME comment above says, we want to kill this function and let the caller get the compaction group (it now passes its cg id), and call cg->as_table_state() directly.  With multiple compaction groups it's wrong for the table to provide as_table_state (although it's ironic, but it's just a relic of the old, per-tablers days)

Reply:
It's not the only one, but there aren't many of them, I will delete this method